com.unity.netcode.gameobjects/Runtime/Serialization/FastBufferReader.cs

```diff
@@ -93,12 +93,12 @@ internal unsafe void CommitBitwiseReads(int amount)
         /// <param name="offset"></param>
         public unsafe FastBufferReader(NativeArray<byte> buffer, Allocator allocator, int length = -1, int offset = 0)
         {
-            Handle = CreateHandle((byte*)buffer.GetUnsafePtr(), Math.Max(1, length == -1 ? buffer.Length : length), offset, allocator);
+            Handle = CreateHandle((byte*)buffer.GetUnsafePtr(), length == -1 ? buffer.Length : length, offset, allocator);
```

Comment:
`length = -1` + `length == -1 ? ... : ...` vs `length = 0` + `length < 1 ? ... : ...`?

can we have length = 0 as an indicator instead of -1, or is that for technical reasons?

Reply:
0 is a valid length. If a user passes in a length of 0, then we should assume they want a 0-length reader. -1 is an indicator to calculate the length based on the input. I think it would be bad to create a situation where creating a 0-length reader is impossible because passing 0 results in length being automatically calculated and ending up non-zero.

Reply:
what is a 0 length reader?

Reply:
What is a 0 length string or buffer? It may not seem immediately useful, but if I receive data from an external source and the length of that data is 0, I should be able to create a buffer reader from that data validly. If I have an input that has 50 bytes in it, the first two bytes are a length of a sub buffer, and those bytes contain a length of 0, if I create `new FastBufferReader(subBuffer, Allocator.Temp, length, 2)`, that shouldn't result in me getting a reader of size 48 if `length` is 0. I should get a reader of size 0.

It's not useful to read from a 0-size buffer, and also not useful to print a 0-length string, but there are many cases where it's useful to know the buffer has 0 length and changing its container shouldn't change that.

Reply:
in a sense, length 0 is TOptional with set = false here — it's not null, it's valid but empty.
it's fine, I was just curious, as I said in the review comment, it's good to go!